keras_core/ops/math.py

```diff
@@ -314,6 +314,66 @@ def qr(x, mode="reduced"):
     return backend.math.qr(x, mode=mode)
 
 
+class Frame(Operation):
+    def __init__(self, frame_length, frame_step):
+        super().__init__()
+        self.frame_length = frame_length
+        self.frame_step = frame_step
+
+    def compute_output_spec(self, x):
+        if len(x.shape) < 1:
+            raise ValueError(
+                f"Input should have rank >= 1. "
+                f"Received: input.shape = {x.shape}"
+            )
+        if x.shape[-1] is not None:
+            num_frames = (
+                1 + (x.shape[-1] - self.frame_length) // self.frame_step
+            )
+        else:
+            num_frames = None
+        new_shape = x.shape[:-1] + (num_frames, self.frame_length)
+        return KerasTensor(shape=new_shape, dtype=x.dtype)
+
+    def call(self, x):
+        return backend.math.frame(
+            x, frame_length=self.frame_length, frame_step=self.frame_step
+        )
+
+
+@keras_core_export("keras_core.ops.frame")
+def frame(x, frame_length, frame_step):
```

Comment:
I'm not 100% sure we should export `frame` to the public API as an op (the function here + the `Frame` class). How important is it compared to the other too?

Reply:
I have discovered that it is used to build LongFormer (citation> 2k) in huggingface implementation
https://github.com/huggingface/transformers/blob/eec5841e9f440c795fb9292d009675d97a14f983/src/transformers/models/longformer/modeling_tf_longformer.py#L1107

There are about 1.6k search results for `tf.signal.frame`
https://github.com/search?q=tf.signal.frame&type=code

I believe that exporting it holds value. However, I agree that we should find a more suitable place for it.

Refs:
- tf: `tf.signal`
- torch: `torch.Tensor.unfold`
- jax & numpy: no direct api

Is a new namespace required for it?

Reply:
Thanks for the info. OK to export. It is like a 1D equivalent of `extract_patches`, right? In this case perhaps we can standardize the API on `extract_frames` or `extract_windows` (which is also more self-explanatory to a user who does not know what it does).

We might also want to standardize the terminology, e.g. `step` -> `stride`. We also need to decide whether to refer to frames as "frames", "windows", or "sequences". A very similar utility exists for tf.data (`timeseries_dataset_from_array`) which uses "sequence".

What do you think?

Reply:
Sounds good to me!

Reply:
Updated.